llrt_core/src/vm.rs

```diff
@@ -67,6 +68,25 @@ pub(crate) static COMPRESSION_DICT: &[u8] =
 static DECOMPRESSOR_DICT: Lazy<DecoderDictionary> =
     Lazy::new(|| DecoderDictionary::copy(COMPRESSION_DICT));
 
+static NODE_MODULES_PATH: Lazy<String> = Lazy::new(|| {
+    let current_dir = env::current_dir()
+        .ok()
+        .and_then(|path| path.into_os_string().into_string().ok())
+        .unwrap_or_else(|| "/".to_string());
+    let mut base_dir = PathBuf::from_str(&current_dir).unwrap();
+
+    loop {
+        let node_modules_path = base_dir.join("node_modules");
+        if node_modules_path.exists() && node_modules_path.is_dir() {
+            return node_modules_path.to_str().unwrap().to_string();
+        }
+        if !base_dir.pop() {
+            break;
+        }
+    }
+    "/node_modules".to_string()
+});
+
```

Comment:
Here, we're just checking if node_modules exists in a base dir or move up the parent directory.
This can't be static as we need each path passed to Custom resolver if any of them contains "node_modules". I think in order to get this right, we must line by line copy the implementation pseudocode mentioned here:
https://nodejs.org/api/modules.html#all-together
Note here that we should probably cache NODE_MODULES_PATHS() functions, return value based on START.
GLOBAL_FOLDERS here would be the paths added to CustomResolver

Reply:
Thank you for pointing this out. This was a major correction, but we believe it is an improvement.

Reply:
Sorry, what I meant was replace the entire resolution implementation to pseudo algorithm I linked. The only difference from node is that we check for ".lrt" in addition to ".js" ."cjs" and ".mjs also core modules are located in bytecode_cache

Reply:
It think they share a lot of similarities (for step 1-5) but worth refactor our current code so its a 1:1 implementation. What I don't understand is 4. When do you ever use "#"? I have never seen it.

Reply:
Certainly, a 1:1 implementation would be easier to compare/validate with pseudo-code, and the code would be more transparent.

Also, I have never seen a module that starts with a '#'. This is why I say that only what is possible should be implemented.